openssl-dynamic/src/main/java/io/netty/internal/tcnative/CertificateCallback.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.internal.tcnative;
+
+/**
+ * Is called during handshake and hooked into openssl via {@code SSL_CTX_set_cert_cb}.
+ *
+ * IMPORTANT: Implementations of this interface should be static as it is stored as a global reference via JNI. This
+ *            means if you use an inner / anonymous class to implement this and also depend on the finalizer of the
+ *            class to free up the SSLContext the finalizer will never run as the object is never GC, due the hard
+ *            reference to the enclosing class. This will most likely result in a memory leak.
+ */
+public interface CertificateCallback {
+
+    /**
+     * The types contained in the {@code keyTypeBytes} array.
+     */
+    // Extracted from https://github.com/openssl/openssl/blob/master/include/openssl/tls1.h
+    byte TLS_CT_RSA_SIGN = 1;
```

Comment:
because we really want to access the byte value later on.

Reply:
I see.   I was thinking (optional of course) something like:

```
enum KeyTypes {
  TLS_CT_RSA_SIGN(1),
  TLS_CT_RSA_FIXED_DH(4);
  // ...
  byte getByte() {
    // ...
  }
}
```

This would allow `handle()` to be an EnumSet, and no risk of unsupported bytes being pased in.

Reply:
I think I will keep it as byte for now to reduce garbage / and keep it simple in JNI